### ISSUE TITLE
Add Skip Link to All Regions

### ISSUE THREAD

**Issue by manfromjupyter** [jupyterlab/jupyterlab#9688]
Requirement: WCAG 2.4.1 (A)

From: #9399 and doesn't require a screenreader to make/test.
> Expectation: The user can skip navigation functions/sidebar and go straight to the content
> - No skip link exists is available to skip the header, navs, etc and just straight to the left panel or notebook, etc editing.
> - Proposed Solution: On first or second tab, have an element that is off screen gain focus and upon gaining focus changes position down 50px or whatever amount (all via CSS) and something that is either a single skip to the notebook panel or a table of contents that lets you jump to sidebar, left panel, notebook panel, whatever.

For now could simply add a single link. Example how: https://css-tricks.com/how-to-create-a-skip-to-content-link/ but ideally if possible, upon tabbing once, it should cause with css because of this focus to move up. THe container itself is probably a `<div>` with `aria-label="Skip to section"` and gets lowered where you then see a list of like 5-7 anchor links that are linked to the ids of each region we put a landmark on from #9622. 
ex.
1. Header
2. Main
3. Primary Sidetray
(May need to add landmark for the Stacked Panel region with id="jp-left-stack")
4. Secondary Sidetray
(May need to add landmark for the Stacked Panel region with id="jp-right-stack")
5. Footer

If they don't have ids need to add one.

Wait until after the landmark tickets get merged so you have landmarks to jump to (https://github.com/jupyterlab/jupyterlab/issues/9491).

**Comment by 0618** [comment]
🙋‍♀️ I can try working on the skip link.

**Comment by manfromjupyter** [comment]
@0618 Yes please! If you get stuck or have questions just reach out here and @isabela-pf and I will tag the people that can help (unless you already know who to talk to). Once you get close I think we can make another ticket with just the info for the skip link in there so it's clear to people what's being merged.

**Comment by isabela-pf** [comment]
@gabalafou was experimenting with the skip link and mentioned some more notes we'll want to be aware of when we get here.

- It seems the current skip link doesn't disappear from view when used, a regression from the work in #10126
- Because we are currently lacking visible focus indicators for all regions of the UI, it can be unclear to users what the skip link did/if it worked because the left side bar is open by default. To be clear, the focus works properly. It may be worth having some kind of stopgap for this if possible.
- Should this be something that can be easily be brought up again rather than only being accessible upon tabbing through the very first layer of JupyterLab? This isn't quite a conventional use of a skip link, but it may be helpful to basically have a shortcut for navigating by keyboard through the various areas of the UI at any point (for example, to switch quickly between a document and a side bar extension contolling it).
